extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -218,17 +229,24 @@ public String getExtensionLibPath(final String extensionName) throws StoreAccess
         }
     }
 
-    public String getExtensionResource(final String resourcePath) throws StoreAccessException {
+    public String getExtensionResource(final String resourcePath) throws FalconException {
         if (StringUtils.isBlank(resourcePath)) {
             throw new StoreAccessException(new Exception("Resource path cannot be null or empty"));
         }
 
         try {
             Path resourceFile = new Path(resourcePath);
+            InputStream data;
 
             ByteArrayOutputStream writer = new ByteArrayOutputStream();
-            InputStream data = fs.open(resourceFile);
-            IOUtils.copyBytes(data, writer, fs.getConf(), true);
+            if (resourcePath.startsWith("hdfs")){
```

Comment:
The scheme could be webhdfs or hftp.. or anything else... A better option would be to check for file:// and assume HDFS in the else part. 

Even better would be to take in the extension name also, see if it is trusted or not and take action appropriately.